### ISSUE TITLE
Floating point exception (FPE) in MOM_MEKE beta calculation

### ISSUE THREAD

**Issue by marshallward** [NOAA-GFDL/MOM6#917]
When running the GNU debug build, there is a floating point exception in the calculation of `beta` in `MEKE_equilibrium` in `MOM_MEKE.F90`:
```
 585     beta = sqrt( ( US%s_to_T*G%dF_dx(i,j) - CS%MEKE_topographic_beta*FatH/G%bathyT(i,j)* &
 586            (G%bathyT(i+1,j) - G%bathyT(i-1,j))/2./G%dxT(i,j) )**2. &
 587            + ( US%s_to_T*G%dF_dy(i,j) - CS%MEKE_topographic_beta*FatH/G%bathyT(i,j) &
 588            *(G%bathyT(i,j+1) - G%bathyT(i,j-1))/2./G%dyT(i,j) )**2. )
```
This was observed in the `Baltic_OM4_025` and `Baltic_OM4_05` tests, and possibly others with higher CPU counts.

The specific FPE was not listed in the error but the most likely explanation (according to @adcroft) is a zero-valued bathymetry with a division by zero.

I can't say why this has not caused problems in production, but the variable does not propagate very far and may disappear in subsequent calculations.

Fixing this is probably not a good idea, since it will probably change answers, so ought to be addressed in the future.

**Comment by marshallward** [comment]
To follow up on this, our test cases all have `CS%MEKE_topographic_beta = 0.`, so we are likely computing `0. * Inf`, which is a `NaN`.  This number is later passed into a `max()` function.

And in the infinite wisdom that is IEEE754, `max(NaN, x) = x`, so this most likely explains why this did not previously raise an error.

I will try modifying this equation to behave more safely, but there is a chance it will change the behavior of the `max` call, and therefore the numbers may change.

**Comment by marshallward** [comment]
Addressed in #919, moving discussion to there.
